File: 프로그래머스/0/181929. 원소들의 곱과 합/원소들의 곱과 합.py
def solution(num_list):
    answer = 0
    answer1 = 1
    answer2 = 0
    
    for i in range(len(num_list)):
        answer1 *= num_list[i]
        answer2 += num_list[i]
        
    answer3 = answer2 ** 2
    
    if answer1 < answer3 :
        answer = 1
    else:
        answer = 0
    
    return answer

File: 프로그래머스/0/181929. 원소들의 곱과 합/원소들의 곱과 합.swift
func solution(_ num_list: [Int]) -> Int {
    var sum = 0
    var mul = 1

    for num in num_list {
        sum += num
        mul *= num
        // 곱이 이미 합의 제곱보다 크면 바로 0 반환
        if mul > sum * sum {
            return 0
        }
    }

    return 1
}
